### ISSUE TITLE
[Bug] [master] On the problem of repeated scheduling of dolphins

### ISSUE THREAD

**Issue by LinYuHeng-Con** [apache/dolphinscheduler#15323]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Dolphins experience repeated scheduling

### What you expected to happen

Schedule only once

### How to reproduce

Accidental appearance，At present, the production tasks are basically scheduled repeatedly every day

### Anything else

![image](https://github.com/apache/dolphinscheduler/assets/48077349/8af33dac-9955-48d8-b84c-d60e791e478d)
![image](https://github.com/apache/dolphinscheduler/assets/48077349/646e779f-4a57-45f4-962c-41fd51804af1)
![image](https://github.com/apache/dolphinscheduler/assets/48077349/e30b0a2d-5eb4-4966-8d70-7acddfbb96d6)
![image](https://github.com/apache/dolphinscheduler/assets/48077349/06a8ca8f-302e-4444-a529-bddbaed8faff)
![image](https://github.com/apache/dolphinscheduler/assets/48077349/b40924e1-37b9-4d48-a49e-ce6b705ab4de)
![image](https://github.com/apache/dolphinscheduler/assets/48077349/0621b2ec-e95a-4ba0-939e-9aee46a0e291)

Currently, dolphins use the LocalDataSourceJobStore class, but due to setDontSetAutoCommitFalse (true), autocommit is enabled, transactions are not enabled, and distributed locks are not effective

I plan to modify the LocalDataSourceJobStore in the configuration file to JobStoreCMT, or customize a JobStore to see the effect

### Version

3.1.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by SbloodyS** [comment]
Would you like to submit a pr to fix this? @LinYuHeng-Con

**Comment by LinYuHeng-Con** [comment]
> Would you like to submit a pr to fix this? @LinYuHeng-Con

Let me roughly describe the problem of repeated scheduling before,

1. As Tidb also supports the MySQL protocol, the data of Dolphinscheduler was previously stored in Tidb, resulting in duplicate scheduling issues

2. So we consider whether the duplicate scheduling problem is caused by Tidb, as Tidb is a distributed database, and we plan to switch Tidb to MySQL

3. However, before switching, I discovered the aforementioned issues and customized a JobStore, but there were still issues with duplicate scheduling

4. So we switched Tidb directly to MySQL, and the configuration still used org. springframework. scheduling. quartz. LocalDataSourceJobStore

5. The problem of repeated scheduling did not occur again afterwards

6. At present, I am unable to determine whether it is due to the database or the code. I believe that there is a problem with the logic of obtaining distributed locks, but in reality, the problem of duplicate scheduling has not occurred again
